src/test/java/com/google/cloud/tools/appengine/cloudsdk/DevAppServerRunnerTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk;
+
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ProcessBuilderFactory;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessHandler;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessHandlerException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DevAppServerRunnerTest {
+
+  private final boolean IS_WINDOWS = System.getProperty("os.name").contains("Windows");
+
+  @Rule public TemporaryFolder testFolder = new TemporaryFolder();
+
+  @Mock private CloudSdk sdk;
+  @Mock private ProcessHandler processHandler;
+  @Mock private ProcessBuilderFactory processBuilderFactory;
+  @Mock private ProcessBuilder processBuilder;
+  @Mock private Process process;
+  @Mock private Map<String, String> processEnv;
+  private Path javaExecutablePath;
+  private Path appengineToolsJar;
+  private Path appengineJavaSdkPath;
+  private File workingDirectory;
```

Comment:
Why is this one a File and the rest Paths? This may be something else to clean up.